include/io/KeyMap.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef CHIP_8_KEYMAP_HPP
+#define CHIP_8_KEYMAP_HPP
+
+#include <array>
+#include <SFML/Window/Keyboard.hpp>
+
+namespace chip8::io
+{
+    class KeyMap
+    {
+        public:
+            static constexpr uint8_t num_keys { 16 };
+            //I felt using an actual std::map for this was overkill
+            using KeyMapping = std::array<sf::Keyboard::Key, num_keys>;
+
+        public:
+            KeyMap();
+            ~KeyMap() = default;
```

Comment:
Is there a reason for making this explicit? Isn't it implied by a user defined constructor?

Reply:
You mean the destructor? It's just out of habit and to point out that I did not forget to include a custom destructor or adding a virtual one (in case anytone wants to inherit from this class).